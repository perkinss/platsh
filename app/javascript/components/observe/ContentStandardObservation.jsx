import React from 'react'
import {Fade} from "@material-ui/core/index";
import { Grid, Tooltip, Typography } from '@material-ui/core'
import StandardCheckbox from './StandardCheckbox'


export default function ContentStandardObservation (props) {
    const { id,  task, studentId, classes, description, isMet, dispatchFormUpdate } = props

    return (
        <React.Fragment key={id}>
            <Grid item key={id} xs={1} xl={1}>
                <StandardCheckbox
                    id={id}
                    taskId={task.id}
                    studentId={studentId}
                    isChecked={isMet}
                    dispatchFormUpdate={dispatchFormUpdate}
                    classes={classes} />
            </Grid>
            <Grid item key={`id-${id}`} xs={11} xl={11}>
                <Tooltip  TransitionComponent={Fade} TransitionProps={{ timeout: 600 }} title={description}>
                    <Typography
                        noWrap={true}
                        paragraph
                        align={'left'}
                        style={{paddingLeft: '10px', marginBottom: 0}}>
                        {description}
                    </Typography>
                </Tooltip>
            </Grid>
        </React.Fragment>
    )
}

