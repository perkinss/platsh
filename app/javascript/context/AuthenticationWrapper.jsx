import React, { useEffect, useState, useContext } from 'react';
import User from '../model/user'

export const RootAuthContext = React.createContext();
export default ({ children }) => {
  let anyAuthVal,anyAuthData;
  try {
    anyAuthVal=JSON.parse(window.sessionStorage.getItem('authenticated'));
  }
  catch(err) {
    anyAuthVal=false;
  }
  try {
    anyAuthData = JSON.parse(window.sessionStorage.getItem('authData'))
  }
  catch(err) {
    anyAuthData = null;
  }
  const guestUser = new User({})
  const prevAuth = (anyAuthVal ? anyAuthVal : false);
  const prevAuthData = (anyAuthData ? anyAuthData : null);
  const prevUser = (anyAuthData ? new User(anyAuthData) : guestUser);
  const [authenticated, setAuthenticated] = useState(prevAuth);
  const [authData, setAuthData] = useState(prevAuthData);
  const [currentUser, setCurrentUser] = useState(prevUser);
  useEffect(
    () => {
      window.sessionStorage.setItem('authenticated', JSON.stringify(authenticated));
      window.sessionStorage.setItem('authData', (authData ? JSON.stringify(authData):null));
      setCurrentUser(authData ? new User(authData) : guestUser)
    },
    [authenticated, authData]
  );
  const defaultContext = {
    authenticated, setAuthenticated,
    authData, setAuthData,
    currentUser, setCurrentUser
  };
  return (
    <RootAuthContext.Provider value={defaultContext}>
      {children}
    </RootAuthContext.Provider>
  );
};
