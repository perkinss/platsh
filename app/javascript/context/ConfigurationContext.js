import React from 'react'

export const configuration = {
    sections: [
        {
            id: null,
            courses: [],
            name: '',
            assessment: [
                {
                    type: 'test',
                    name: '',
                    tasks: [
                    {
                        name: '',
                        available_standards: [
                            {
                                id: -1,
                                level: ''
                            }
                        ],
                        competencies: [

                        ]
                    },
                    ]
                }
            ],
            enrollees: [
                1, 2, 3, 22
            ]
        },
    ],

}


const ConfigurationContext =  React.createContext(configuration);

export { ConfigurationContext }
