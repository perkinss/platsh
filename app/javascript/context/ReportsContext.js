import React from 'react'

export const reports = {
    students: [],
    sections: [],
    selected_section: null,
    selected_student: null,
    section_overview_marks: {},
    section_competency_scores: {},
    student_section_marks: {},
    heat_map_colors: []
}


const ReportsContext =  React.createContext(reports);

export { ReportsContext }
