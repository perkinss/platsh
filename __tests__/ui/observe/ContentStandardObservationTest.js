import React from 'react'
import { createRender } from '@material-ui/core/test-utils';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect'
import ContentStandardObservation from "../../../app/javascript/components/observe/ContentStandardObservation";
import {ConfigurationContext} from "../../../app/javascript/context/ConfigurationContext"

Enzyme.configure({ adapter: new Adapter() });

describe('<ContentStandardObservation />', () => {
    let render;
    let question = {
        id: 1,
        name: 'Question 1',
        contentStandards: [
            1, 2, 3,
        ],
        levels: [
            1, 0, 2
        ],
        competencies: [
            2
        ],
        schemas: [1]
    }

    const classes = {}
    const dispatch = () => {}
    const configuration = {
        student_marks: {
            1: {1: {1: 1}}
        }
    }

    beforeAll(() => {
        render = createRender();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ConfigurationContext.Provider value={{dispatch: dispatch, state: configuration}} >
                <ContentStandardObservation task={question} classes={classes} description={"test"}/>
            </ConfigurationContext.Provider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });

    // Test whether clicking the add button opens the course-section panel for the selected course

})


/**import { configuration, ConfigurationContext } from "../../app/javascript/context/ConfigurationContext";

 Enzyme.configure({ adapter: new Adapter() });

 describe('<StudentHeader />', () => {
    let render;
    const dispatch = () => {}
    const configuration = {}

    beforeAll(() => {
        render = createShallow();
    })

    it('should match the snapshot', () => {
        const wrapper = render(
            <ConfigurationContext.Provider value={{dispatch: dispatch, state: configuration}} >
            <StudentHeader topic={"A topic"} competency={"Fantastico"} id={1} classes={{}}/>
            </ConfigurationContext.Provider>);
        expect(wrapper).not.toBeNull()
        expect(wrapper).toMatchSnapshot()
    });


 **/