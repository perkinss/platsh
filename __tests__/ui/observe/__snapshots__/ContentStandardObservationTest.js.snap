// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentStandardObservation /> should match the snapshot 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-xl-1",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "aria-disabled": "false",
          "class": "MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "MuiIconButton-label",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "PrivateSwitchBase-input-4",
                  "data-indeterminate": "false",
                  "tabindex": "-1",
                  "type": "checkbox",
                },
                "children": Array [],
                "name": "input",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "MuiSvgIcon-root",
                    "focusable": "false",
                    "viewBox": "0 0 24 24",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "focusable": undefined,
                    "viewBox": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "focusable": undefined,
                    "viewBox": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-indeterminate": undefined,
                  "tabindex": undefined,
                  "type": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-indeterminate": undefined,
                  "tabindex": undefined,
                  "type": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "MuiSvgIcon-root",
                  "focusable": "false",
                  "viewBox": "0 0 24 24",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "PrivateSwitchBase-input-4",
                    "data-indeterminate": "false",
                    "tabindex": "-1",
                    "type": "checkbox",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-indeterminate": undefined,
                    "tabindex": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-indeterminate": undefined,
                    "tabindex": undefined,
                    "type": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "focusable": undefined,
                  "viewBox": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "focusable": undefined,
                  "viewBox": undefined,
                },
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-disabled": undefined,
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-disabled": undefined,
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Node {
      "attribs": Object {
        "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-xl-11",
      },
      "children": Array [
        Node {
          "attribs": Object {
            "class": "MuiTypography-root MuiTypography-body1 MuiTypography-noWrap MuiTypography-paragraph MuiTypography-alignLeft",
            "style": "padding-left:10px;margin-bottom:0",
            "title": "test",
          },
          "children": Array [
            Node {
              "data": "test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "p",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
            "title": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
            "title": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": [Circular],
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "parent": Node {
      "children": Array [
        [Circular],
        Node {
          "attribs": Object {
            "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-xl-11",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "MuiTypography-root MuiTypography-body1 MuiTypography-noWrap MuiTypography-paragraph MuiTypography-alignLeft",
                "style": "padding-left:10px;margin-bottom:0",
                "title": "test",
              },
              "children": Array [
                Node {
                  "data": "test",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "p",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
                "title": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
                "title": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "1": Node {
    "attribs": Object {
      "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11 MuiGrid-grid-xl-11",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiTypography-root MuiTypography-body1 MuiTypography-noWrap MuiTypography-paragraph MuiTypography-alignLeft",
          "style": "padding-left:10px;margin-bottom:0",
          "title": "test",
        },
        "children": Array [
          Node {
            "data": "test",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "p",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
          "title": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        Node {
          "attribs": Object {
            "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-xl-1",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "aria-disabled": "false",
                "class": "MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "MuiIconButton-label",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "class": "PrivateSwitchBase-input-4",
                        "data-indeterminate": "false",
                        "tabindex": "-1",
                        "type": "checkbox",
                      },
                      "children": Array [],
                      "name": "input",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Node {
                        "attribs": Object {
                          "aria-hidden": "true",
                          "class": "MuiSvgIcon-root",
                          "focusable": "false",
                          "viewBox": "0 0 24 24",
                        },
                        "children": Array [
                          Node {
                            "attribs": Object {
                              "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                            },
                            "children": Array [],
                            "name": "path",
                            "namespace": "http://www.w3.org/2000/svg",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "d": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "d": undefined,
                            },
                          },
                        ],
                        "name": "svg",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "viewBox": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "aria-hidden": undefined,
                          "class": undefined,
                          "focusable": undefined,
                          "viewBox": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "data-indeterminate": undefined,
                        "tabindex": undefined,
                        "type": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "data-indeterminate": undefined,
                        "tabindex": undefined,
                        "type": undefined,
                      },
                    },
                    Node {
                      "attribs": Object {
                        "aria-hidden": "true",
                        "class": "MuiSvgIcon-root",
                        "focusable": "false",
                        "viewBox": "0 0 24 24",
                      },
                      "children": Array [
                        Node {
                          "attribs": Object {
                            "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                          },
                          "children": Array [],
                          "name": "path",
                          "namespace": "http://www.w3.org/2000/svg",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "d": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "d": undefined,
                          },
                        },
                      ],
                      "name": "svg",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": Node {
                        "attribs": Object {
                          "class": "PrivateSwitchBase-input-4",
                          "data-indeterminate": "false",
                          "tabindex": "-1",
                          "type": "checkbox",
                        },
                        "children": Array [],
                        "name": "input",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "data-indeterminate": undefined,
                          "tabindex": undefined,
                          "type": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "data-indeterminate": undefined,
                          "tabindex": undefined,
                          "type": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                        "focusable": undefined,
                        "viewBox": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "aria-hidden": undefined,
                        "class": undefined,
                        "focusable": undefined,
                        "viewBox": undefined,
                      },
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-disabled": undefined,
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-disabled": undefined,
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": Node {
      "attribs": Object {
        "class": "MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-xl-1",
      },
      "children": Array [
        Node {
          "attribs": Object {
            "aria-disabled": "false",
            "class": "MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "MuiIconButton-label",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "PrivateSwitchBase-input-4",
                    "data-indeterminate": "false",
                    "tabindex": "-1",
                    "type": "checkbox",
                  },
                  "children": Array [],
                  "name": "input",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "aria-hidden": "true",
                      "class": "MuiSvgIcon-root",
                      "focusable": "false",
                      "viewBox": "0 0 24 24",
                    },
                    "children": Array [
                      Node {
                        "attribs": Object {
                          "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                        },
                        "children": Array [],
                        "name": "path",
                        "namespace": "http://www.w3.org/2000/svg",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "d": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "d": undefined,
                        },
                      },
                    ],
                    "name": "svg",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                      "focusable": undefined,
                      "viewBox": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "aria-hidden": undefined,
                      "class": undefined,
                      "focusable": undefined,
                      "viewBox": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-indeterminate": undefined,
                    "tabindex": undefined,
                    "type": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-indeterminate": undefined,
                    "tabindex": undefined,
                    "type": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "MuiSvgIcon-root",
                    "focusable": "false",
                    "viewBox": "0 0 24 24",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "d": "M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "PrivateSwitchBase-input-4",
                      "data-indeterminate": "false",
                      "tabindex": "-1",
                      "type": "checkbox",
                    },
                    "children": Array [],
                    "name": "input",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "data-indeterminate": undefined,
                      "tabindex": undefined,
                      "type": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "data-indeterminate": undefined,
                      "tabindex": undefined,
                      "type": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "focusable": undefined,
                    "viewBox": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "focusable": undefined,
                    "viewBox": undefined,
                  },
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-disabled": undefined,
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-disabled": undefined,
            "class": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": Node {
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
      },
      "prev": null,
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
      },
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
